synapse/app/__init__.py

```diff
@@ -12,3 +12,23 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+import sys
+sys.dont_write_bytecode = True
+
+from synapse.python_dependencies import (
+    check_requirements, MissingRequirementError
+)  # NOQA
+
+if __name__ == '__main__':
```

Comment:
It seems rude to sys.exit() unless you being run as the main program, but I don't think it should cause any problems for any of the likely ways this code will be run.